optaplanner-core/src/main/java/org/optaplanner/core/impl/phase/scope/AbstractMoveScope.java

```diff
@@ -48,11 +48,11 @@ public int getMoveIndex() {
         return move;
     }
 
-    public <Score_ extends Score<Score_>> Score_ getScore() {
-        return (Score_) score;
+    public Score getScore() {
+        return score;
     }
 
-    public <Score_ extends Score<Score_>> void setScore(Score_ score) {
+    public void setScore(Score score) {
         this.score = score;
     }
```

Comment:
This is unrelated, but these generics were pointless and were causing unnecessary warnings in some tests.